app/src/main/java/de/danoeh/antennapod/dialog/IntraFeedSortDialog.java

```diff
@@ -16,35 +16,59 @@
     @NonNull
     protected Context context;
 
-    public IntraFeedSortDialog(@NonNull Context context, @Nullable SortOrder sortOrder) {
+    final private String[] sortItems;
+    final private SortOrder[] sortValues;
+
+    public IntraFeedSortDialog(@NonNull Context context, @Nullable SortOrder sortOrder, @NonNull boolean isLocalFeed) {
         this.context = context;
         this.currentSortOrder = sortOrder;
+
+        if (isLocalFeed) {
+            sortItems = context.getResources().getStringArray(R.array.local_feed_episodes_sort_options);
+            final String[] localSortStringValues =
+                    context.getResources().getStringArray(R.array.local_feed_episodes_sort_values);
+            sortValues = SortOrder.valuesOf(localSortStringValues);
+        } else {
+            sortItems = context.getResources().getStringArray(R.array.feed_episodes_sort_options);
+            final String[] commonSortStringValues =
+                    context.getResources().getStringArray(R.array.feed_episodes_sort_values);
+            sortValues = SortOrder.valuesOf(commonSortStringValues);
+        }
     }
 
     public void openDialog() {
-        final String[] items = context.getResources().getStringArray(R.array.feed_episodes_sort_options);
-        final String[] valueStrs = context.getResources().getStringArray(R.array.feed_episodes_sort_values);
-        final SortOrder[] values = new SortOrder[valueStrs.length];
-        for (int i = 0; i < valueStrs.length; i++) {
-            values[i] = SortOrder.valueOf(valueStrs[i]);
-        }
+        final SortOrder[] values = sortValues;
+        final String [] items = sortItems;
+
+        int idxCurrentSort = getCurrentSortOrderIndex(values);
```

Comment:
When opening the sort dialog of a new subscription that never had a sort value selected, the dialog does not show a selection. Instead, it should show the first item (date new -> old) as selected because that is the default sort order.

Reply:
Makes sense. I decided to return 0 as the default sort order if a sort order is not found. Let me know if I should return -1 instead and check it here instead.